src/libsystemd/sd-journal/journal-vacuum.c

```diff
@@ -151,7 +151,8 @@ int journal_directory_vacuum(
 
                 unsigned long long seqnum = 0, realtime;
                 _cleanup_free_ char *p = NULL;
-                sd_id128_t seqnum_id;
+                /* If the journal file is archived, seqnum_id won't be initialised before use, so set to 0 */
+                sd_id128_t seqnum_id = SD_ID128_MAKE(00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00);
```

Comment:
SD_ID128_NULL

Reply:
maybe move the initialization with SD_ID128_NULL into the if block that handles archived journal files though. After all for non-archived ones this initialization is unnecessary

Reply:
I knew there was something already. Went and checked - completely missed it anyway :-D

Reply:
moved down and used SD_ID128_NULL